spark/src/main/scala/org/apache/spark/sql/delta/commands/DescribeDeltaHistoryCommand.scala

```diff
@@ -37,15 +59,38 @@ import org.apache.spark.sql.execution.command.LeafRunnableCommand
  * @param options: Hadoop file system options used for read and write.
  */
 case class DescribeDeltaHistory(
-    path: Option[String],
-    tableIdentifier: Option[TableIdentifier],
+    override val child: LogicalPlan,
     limit: Option[Int],
-    options: Map[String, String],
-    output: Seq[Attribute] = ExpressionEncoder[DeltaHistory]().schema.toAttributes)
-  extends LeafNode with MultiInstanceRelation  {
-  override def computeStats(): Statistics = Statistics(sizeInBytes = conf.defaultSizeInBytes)
-
+    options: Map[String, String] = Map.empty,
+    override val output: Seq[Attribute] = DescribeDeltaHistory.schema.toAttributes)
+  extends UnaryNode with MultiInstanceRelation with DeltaCommand {
   override def newInstance(): LogicalPlan = copy(output = output.map(_.newInstance()))
+
+  override def withNewChildInternal(newChild: LogicalPlan): LogicalPlan = copy(child = newChild)
+
+  /**
+   * Define this operator as having no attributes provided by children in order to prevent column
+   * pruning from trying to insert projections above the source relation.
+   */
+  override lazy val references: AttributeSet = AttributeSet.empty
+
+  override def inputSet: AttributeSet = AttributeSet.empty
+
+  assert(!child.isInstanceOf[Project],
+    s"The child operator of DescribeDeltaHistory must not contain any projection: $child")
+
+  def toCommand: DescribeDeltaHistoryCommand = {
+    import DescribeDeltaHistory.commandName
+    val tableMetadata: Option[CatalogTable] = getTableCatalogTable(child)
```

Comment:
This can use `getDeltaTable` instead to get the V2 Delta table, and then extract the CatalogTable from there. This will reduce the indirection from `getTableCatalogTable`.